### PR TITLE
TBOX-301: create tamr client using jwt token auth

### PR DESCRIPTION
**Pull request by zbpvarun-tamr**
# ↪️ Pull Request

## ✔️ PR Todo

- [x] Add documentation
- [ ] Add examples
- [x] Ensure you are following the practices in the [contributor guide](https://docs.google.com/document/d/1c_hhpDMhO0zN793Z_HFOz42Q4g7l1nPRVQ5pXOUsMJU/edit) and [coding_standards](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards)
  * [Use consistent and descriptive name conventions](https://docs.google.com/document/d/1c_hhpDMhO0zN793Z_HFOz42Q4g7l1nPRVQ5pXOUsMJU/edit#heading=h.477s36w4rds9) 
  * [Uses google-style docstrings](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Google-style-docstrings)
  * [Uses versioned API/client methods only](https://docs.google.com/document/d/1c_hhpDMhO0zN793Z_HFOz42Q4g7l1nPRVQ5pXOUsMJU/edit#heading=h.yv6df99fyrq2)
  * [Avoid global variables](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Avoid-global-variables)
  * [Avoid giant try/catch](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Avoid-giant-try-/-except)
  * [Don't use mutable default variables](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Don%E2%80%99t-use-mutable-default-arguments)
  * [Use a `main()` function in your scripts](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Use-a-%E2%80%9Cmain()%E2%80%9D--function-in-your-scripts)
- [x] Added/updated testing for this change (ensure you are testing any changes/new code!)
- [x] Update title to link to TBOX issue (i.e. start pull request title with TBOX-xx) 
- [x] Approval from first code-reviewer
- [ ] Approval from second code-reviewer (only required if a second reviewer has left comments)
- [x] Pass all checks

### CODE REVIEW

**Review by wpmartin — DISMISSED** [review]
Functions like the existing client create function: even with invalid/blank credentials, as long as the auth parameter is set the version and health check will pass as these APIs do not require credentials.

File: tamr_toolbox/utils/client.py [hunk 1]
```diff
@@ -85,6 +90,61 @@ def create(
     return client
```
Comment (wpmartin) [inline tamr_toolbox/utils/client.py:92]: Can you add the version check here, like with movement.py and jobs.py? i.e. @requires_tamr_version(min_version="2020.004.0")
def _run_custom(

File: tamr_toolbox/utils/client.py [hunk 2]
```diff
@@ -85,6 +90,61 @@ def create(
...
+    base_path: str = "/api/versioned/v1/",
+    session: Optional[requests.Session] = None,
+    store_auth_cookie: bool = False,
+    enforce_healthy: bool = False,
+) -> Client:
+    """Creates a Tamr client from the provided configuration values using a JWT token instead
+    of a username and password. Note that this feature is only available on v2022.010.0 or later.
+
+    Args:
+        token: A JWT token to authenticate the client
```
Comment (wpmartin) [inline tamr_toolbox/utils/client.py:109]: what is the behaviour if the token is an empty string?

**Review by wpmartin — COMMENTED** [review]
left some comments

File: tamr_toolbox/utils/client.py (hunk 1 above)
Comment (zbpvarun-tamr) [reply tamr_toolbox/utils/client.py:92]: Changed accordingly

File: tamr_toolbox/utils/client.py (hunk 2 above)
Comment (zbpvarun-tamr) [reply tamr_toolbox/utils/client.py:109]: The client will still be created. Users will run into issues though when they try to hit an API that requires credentials.

**Review by wpmartin — APPROVED** [review]
lgtm

**Pull request merged**